app/boards/shields/zmk_uno/zmk_uno.overlay

```diff
@@ -0,0 +1,176 @@
+/*
+ * Copyright (c) 2022 The ZMK Contributors
+ *
+ * SPDX-License-Identifier: MIT
+ */
+
+#include <dt-bindings/led/led.h>
+#include <dt-bindings/zmk/matrix_transform.h>
+
+&arduino_i2c {
+	status = "okay";
+};
+
+&arduino_spi {
+	status = "okay";
+
+	led_strip: ws2812@0 {
+		compatible = "worldsemi,ws2812-spi";
+		label = "RGB";
+
+		/* SPI */
+		reg = <0>; /* ignored, but necessary for SPI bindings */
+		spi-max-frequency = <4000000>;
+
+		/* WS2812 */
+		chain-length = <7>; /* 4 underglow + 3 per-key LEDs */
+		spi-one-frame = <0x70>;
+		spi-zero-frame = <0x40>;
+
+		color-mapping = <LED_COLOR_ID_GREEN LED_COLOR_ID_RED LED_COLOR_ID_BLUE>;
+	};
+};
+
+/ {
+	chosen {
+		zmk,kscan = &kscan_matrix_comp;
+		zmk,backlight = &backlight;
+		zmk,underglow = &led_strip;
+		zmk,matrix-transform = &matrix_transform;
+	};
+
+	// Commented out until we add more powerful power domain support
+	// external_power {
+	// 	compatible = "zmk,ext-power-generic";
+	// 	label = "EXT_POWER";
+	// 	init-delay-ms = <200>;
+	// 	control-gpios = <&arduino_header 1 GPIO_ACTIVE_LOW>;
+	// };
+
+	rgb_power {
+		compatible = "zmk,ext-power-generic";
+		label = "EXT_POWER";
+		// label = "RGB_POWER";
+		init-delay-ms = <200>;
+		control-gpios = <&arduino_header 1 GPIO_ACTIVE_LOW>;
+	};
+
+	backlight: gpioleds {
+		compatible = "gpio-leds";
+		label = "Backlight LEDs";
+		gpio_led_0 {
+			gpios = <&arduino_header 12 GPIO_ACTIVE_HIGH>;
+			label = "Backlight LED 0";
+		};
+	};
+
+	matrix_transform: matrix_transform {
+		compatible = "zmk,matrix-transform";
+		rows = <3>;
+		columns = <4>;
+
+		map = <
+		RC(0,0) RC(0,1)
+		RC(1,0) RC(1,1)
+		RC(2,0) RC(2,1) RC(2,2)
+		>;
+
+
+	};
+
+	kscan_matrix_comp: kscan_matrix_comp {
+		compatible = "zmk,kscan-composite";
+		rows = <1>;
+		columns = <7>;
+
+		label = "KSCAN_MATRIX_COMP";
+
+		matrix {
+			kscan = <&kscan_matrix>;
+		};
+
+		toggle {
+			kscan = <&kscan_sp3t_toggle>;
+			row-offset = <2>;
+		};
+
+	};
+
+	kscan_direct_comp: kscan_direct_comp {
+		compatible = "zmk,kscan-composite";
+
+		label = "KSCAN_DIRECT_COMP";
+		status = "disabled";
```

Comment:
Would it make more sense to set all of these as initially disabled, and then add `status = "okay"` blocks to the keymap, with the direct pin ones commented out and a comment saying to uncomment them and comment out the others to switch to direct pin?

Alternatively, you could leave this as-is, and then have a single block in the keymap to uncomment which sets `status = "disabled"` for the matrix and `status = "okay"` for the direct pin.

Reply:
Added a commented out section for this.